File: Fundamentals_of_Software_Engineering/front/lk/src/main/java/org/suai/common/Repository/LoginRepository.java
package org.suai.common.Repository;

import org.suai.common.Response;
import org.suai.common.enum_.REQUEST_TYPE;
import org.suai.common.model.systm.loginModel;
import org.suai.common.net.Net;


public class LoginRepository {

    public static Response findLogin(loginModel data){
        return new Response(Net.INSTANCE.request(REQUEST_TYPE.POST, "Login/Auth", data));
    }
}

File: Fundamentals_of_Software_Engineering/front/lk/src/main/kotlin/org/suai/common/LoginRepository.kt
package org.suai.common

import java.io.IOException
import org.suai.common.model.systm.loginModel
import org.suai.common.net.Net
import org.suai.common.enum_.REQUEST_TYPE

object LoginRepository {
    fun getResponse(data: loginModel): Response {
        return try {
            Response(sendLoginData(data))
        } catch (e: Exception) {
            e.printStackTrace()
            Response("Throw exception")
        }
    }

    @Throws(IOException::class, InterruptedException::class)
    fun sendLoginData(data: loginModel) = Net.request(REQUEST_TYPE.POST, "Login/Auth", """{"login":"${data.login}", "password":"${data.password}"}""")
}